### DISCUSSION TITLE
BaseCase and input[type="file"]

### DISCUSSION THREAD

**Discussion by Saorhe** [seleniumbase/SeleniumBase#2307] [Q&A] [answered]
I am trying to set up several automations and I am encountering two problems on two different web pages regarding the input to upload a doc.

My code is as follows:

`       
        selector = 'input[type="file"]'

        if self.is_element_present(selector):
            print('-----------PRESENT-----------')
            
        if self.is_element_visible(selector):
            print('-----------VISIBLE-----------')

        if self.is_element_enabled(selector):
            print('-----------ENABLED-----------')  

        if self.show_element(selector):
            print('-----------SHOW-----------')  

        if self.is_element_clickable(selector):
            print('-----------CLICKCLICK-----------')  

        if self.is_element_in_an_iframe(selector):
            print('-----------IFRAME-----------')
            
        self.send_keys(selector,data['documents']['passport'])
`

In this case only the PRESENT and ENABLED are printed and I always receive the error:

 Element {input[type="file"]} was not visible after 7 seconds! 

Not only when I do the send_keys but also when I do self.assert_element.

The input is not inside any frame or shadow-root. I have tried accessing the input by its ID, Xpath, class etc etc. The strange thing is that I have no problem uploading the document with php webdriver selenium, ui vision or the selenium python.

I have tried the demos or uploading files to the following websites and they go perfectly:
https://tus.io/demo
https://cgi-lib.berkeley.edu/ex/fup.html
https://seleniumbase.io/w3schools/file_upload

Has anyone found themselves in the same situation?

**Comment by Saorhe** [comment]
For inexperienced people like me I leave the solution:

Just do the sendkeys with the following:

element = self.driver.find_element('css selector', 'input[type="file"]')
element.send_keys('C:\\img.jpg')

**Comment by mdmintz** [comment, marked as answer]
Sometimes there's a hidden file input field on a webpage. (Eg. `'input[type="file"]'`)

SeleniumBase has a method for revealing those hidden file input fields:

```python
self.show_file_choosers()
```

Here's a method for selecting a file to upload on that input field:

```python
self.choose_file(selector, file_path)
```

Here's an example test that uses that:

[SeleniumBase/examples/test_show_file_choosers.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_show_file_choosers.py)
